.github/workflows/pr-build.yml

```diff
@@ -0,0 +1,78 @@
+name: build and validate
+on:
+  push:
+    branches:
+      - main
+  pull_request_target:
+    branches: 
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    steps:
+       # Set fetch-depth: 0 to fetch commit history and tags for use in version calculation
+      - name: Check out code
+        uses: actions/checkout@v2.3.4
+        with:
+          ref: ${{github.event.pull_request.head.ref}}
+          repository: ${{github.event.pull_request.head.repo.full_name}}
+          fetch-depth: 0
+      
+      - name: create checksum file
+        uses: hypertrace/actions/checksum@main
+
+      - name: Cache packages
+        id: cache-packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.gradle
+          key: gradle-packages-${{ runner.os }}-${{ github.job }}-${{ hashFiles('**/checksum.txt') }}
+          restore-keys: |
+            gradle-packages-${{ runner.os }}-${{ github.job }}
+            gradle-packages-${{ runner.os }}
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_READ_USER }}
+          password: ${{ secrets.DOCKERHUB_READ_TOKEN }}
+      
+      - name: Build with Gradle
+        run: ./gradlew build dockerBuildImages
+
+  validate-helm-charts:
+    runs-on: ubuntu-20.04
+    container: 
+      image: hypertrace/helm-gcs-packager:0.3.1
```

Comment:
just a thought - could we move this container inside the our validate-charts + package-charts actions?

Reply:
We can but I have some doubts about it. check this out: https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/creating-a-docker-container-action

Reply:
What are the doubts? I used a docker containerized action as part of https://github.com/hypertrace/hypertrace-ui/commit/20072af17d37c0f92ea79dd9b212457e5e63d204#diff-bc668a2c9f2299cef15b222055b4b4d5311646caec2e7610e540cee18ae9b948R85-R90 (you can use the action directly and let it build it each time, or just reference the container) and it's pretty easy on the consumer side. Not sure how much of a pain it is to produce/maintain.

Reply:
will do this.

Reply:
also tested here: hypertrace/attribute-service#65

Reply:
If we run validate charts as container action, I would suggest we do to publish charts action - `hypertrace/actions/helm-gcs-publish` as well as container actions if there are no challenges for consistency.